pkg/resourcegroup/controllers/root/root_controller_test.go

```diff
@@ -196,3 +196,98 @@ func TestRootReconciler(t *testing.T) {
 	time.Sleep(time.Second)
 	assert.Equal(t, "group0", e.Object.GetName())
 }
+
+func TestOwnedByConfigSyncPredicate(t *testing.T) {
+	type testCase struct {
+		name string
+		got  bool
+		want bool
+	}
+
+	ownedByConfigSync := &v1alpha1.ResourceGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"app.kubernetes.io/managed-by": "configmanagement.gke.io",
+			},
+			Namespace: "foo",
+			Name:      "bar",
+		},
+	}
+	ownedByConfigSync.SetGroupVersionKind(v1alpha1.SchemeGroupVersion.WithKind("ResourceGroup"))
+
+	ownedBySomeoneElse := &v1alpha1.ResourceGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"app.kubernetes.io/managed-by": "someone-else",
+			},
+			Namespace: "foo",
+			Name:      "bar",
+		},
+	}
+	ownedBySomeoneElse.SetGroupVersionKind(v1alpha1.SchemeGroupVersion.WithKind("ResourceGroup"))
+
+	notOwned := &v1alpha1.ResourceGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "foo",
+			Name:      "bar",
+		},
+	}
+	notOwned.SetGroupVersionKind(v1alpha1.SchemeGroupVersion.WithKind("ResourceGroup"))
+
+	create := func(o client.Object) bool {
+		return OwnedByConfigSyncPredicate{}.Create(
+			event.TypedCreateEvent[client.Object]{Object: o},
+		)
+	}
+	update := func(o client.Object, n client.Object) bool {
+		return OwnedByConfigSyncPredicate{}.Update(
+			event.TypedUpdateEvent[client.Object]{ObjectOld: o, ObjectNew: n},
+		)
+	}
+
+	// not allowed to redefine delete, so we use a different name
+	yeet := func(o client.Object) bool {
+		return OwnedByConfigSyncPredicate{}.Delete(
+			event.TypedDeleteEvent[client.Object]{Object: o},
+		)
+	}
+
+	generic := func(o client.Object) bool {
+		return OwnedByConfigSyncPredicate{}.Generic(
+			event.TypedGenericEvent[client.Object]{Object: o},
+		)
+	}
+
+	testCases := []testCase{
```

Comment:
Could also use some test cases for GVK's that are not resource group, since that is part of the predicate's checks

Reply:
That's implicitly tested elsewhere (which is how I realized that the condition is needed) but added some in dc3459d0.